docs/api-reference.md

```diff
@@ -110,8 +100,21 @@ function ExampleComponent:didMount()
 end
 ```
 
+Alternatively, you can assign it to a function instead:
+```lua
+Roact.createElement("Frame", {
+	-- The provided function will be called whenever the rendered instance changes.
+	[Roact.Ref] = function(rbx)
+		print("Roblox Instance", rbx)
+	end,
+})
+```
+
+!!! warning
+	When `Roact.Ref` is given a funciton, Roact does not guarantee when this function will be run relative to the reconciliation of other props. If you try to read a Roblox property that's being set via a Roact prop, you won't know if you're reading it before or after Roact reconciles that prop!
```

Comment:
In the current reconciler implementation, this isn't actually true. Refs are applied only after all other props are set.

I don't know if we want to make that guarantee, or even how functional refs should have guarantees going forward. I keep thinking more and more that they should be soft-deprecated and eventually removed.

Reply:
Right, I forgot that this part of the behavior changed. What's odd is that with function refs, you end up with the old one getting called with nil and the new one getting called with the new object. And that'll technically happen every time it re-renders, even if the underlying object doesn't change.

I'm kind of in agreement that I generally want to see the old style of ref go away, because it felt like it was leading towards unsafe uses. What I want here in the docs is some sort of warning that gets at *why* we have negative opinions about it, but I couldn't find a good way to phrase it.

Reply:
We can worry about that soft/hard deprecation later I think, just putting `createRef` first is a big step forward.